### ISSUE TITLE
Add an analyzer to warn about redundant null forgiving operators

### ISSUE THREAD

**Issue by hez2010** [dotnet/roslyn#68204]
**Brief description:**

In some code we used to suppress some nullable warnings using null forgiving operators because the compiler was not able to detect some patterns that can prove a variable is not null. But some of those null forgiving operators are no longer necessary due to the continuously improved nullable analysis from the compiler.

So, we want to remove all those unneeded null forgiving operators to improve the nullable safety, which can prevent some changes in other places resulting in new nullable issues around those code.

But currently it is really hard to do so because we have to repeat the work again and again that deletes every null forgiving to see if there are any new warnings.

**Languages applicable:**

C#

**Code example that the analyzer should report:**

```csharp
void Foo(Bar? value)
{
    if (value?.X ?? false)
    {
        Console.WriteLine(value!.X!.Value); // the two null forgiving operators are unnucessary
    }
}

record Bar(bool? X);
```

**Alternatives**

Make this a language feature instead of an analyzer, because nullable analysis of reference nullable types is a language feature.

**Comment by RikkiGibson** [comment]
Thank you opening this suggestion. It looks like this is a duplicate of #25372, so will close in favor of that.
